spec/javascript/apiService.test.js

```diff
@@ -1,78 +1,97 @@
 import apiService from '~/apiService'
 import { request } from '~/api/request'
 
+const mockLeasePostFn = jest.fn(() => Promise.resolve({lease: true}))
+const mockLeasePutFn = jest.fn(() => Promise.resolve({lease: true}))
 const mockPostFn = jest.fn(() => Promise.resolve(true))
 const mockPutFn = jest.fn(() => Promise.resolve(true))
 const mockDestroyFn = jest.fn(() => Promise.resolve(true))
-request.put = mockPutFn
-request.post = mockPostFn
 request.destroy = mockDestroyFn
 
 describe('apiService', () => {
   const fakeAppId = 'fake_application_id'
 
   describe('createOrUpdateLease', () => {
+    beforeAll(() => {
+      request.post = mockLeasePostFn
+      request.put = mockLeasePutFn
+    })
     const fakeLeaseId = 'fake_lease_id'
 
-    test('should submit put request to update lease if id is provided', () => {
+    test('should submit put request to update lease if id is provided', async () => {
       var lease = {
         'id': fakeLeaseId,
         'monthly_parking_rent': 100
       }
       var expectedData = {'lease': lease}
 
-      var result = apiService.createOrUpdateLease(lease, fakeAppId)
-
-      expect(result).resolves.toEqual(true)
-      expect(mockPutFn.mock.calls.length).toEqual(1)
-      expect(mockPutFn.mock.calls[0]).toEqual([`/applications/${fakeAppId}/leases/${fakeLeaseId}`, expectedData])
+      var result = await apiService.createOrUpdateLease(lease, fakeAppId)
+      // apiService.createOrUpdateLease has an async call
+      // that we then destructure and return instead of
+      // returning the result of the async call directly
+      // which returns undefined here instead of true
+      expect(result).toEqual(true)
+      expect(mockLeasePutFn.mock.calls.length).toEqual(1)
+      expect(mockLeasePutFn.mock.calls[0]).toEqual([`/applications/${fakeAppId}/leases/${fakeLeaseId}`, expectedData])
     })
 
-    test('should submit post request to create lease if no id is provided', () => {
+    test('should submit post request to create lease if no id is provided', async () => {
       var lease = {
         'monthly_parking_rent': 100
       }
       var expectedData = {'lease': lease}
 
-      var result = apiService.createOrUpdateLease(lease, fakeAppId)
-      expect(result).resolves.toEqual(true)
-      expect(mockPostFn.mock.calls.length).toEqual(1)
-      expect(mockPostFn.mock.calls[0]).toEqual([`/applications/${fakeAppId}/leases`, expectedData])
+      var result = await apiService.createOrUpdateLease(lease, fakeAppId)
+      // apiService.createOrUpdateLease has an async call
+      // that we then destructure and return instead of
+      // returning the result of the async call directly
+      // which returns undefined here instead of true
```

Comment:
I think we can get rid of this comment now right?

Reply:
yep, removed